examples/basic/variables.tf

```diff
@@ -0,0 +1,252 @@
+  default = {
+    "name" = "bkp_net"
+    "cidr" = "10.52.0.0/24"
+  }
+}
+
+variable "transit_gateway_name" {
+  description = "Name of the existing transit gateway. Existing name must be provided when you want to create new cloud connections."
+  type        = string
+  default     = null
+}
+
+variable "reuse_cloud_connections" {
+  description = "When the value is true, cloud connections will be reused (and is already attached to Transit gateway)"
+  type        = bool
+  default     = true
+}
+
+variable "cloud_connection_count" {
+  description = "Required number of Cloud connections which will be created/Reused. Maximum is 2 per location"
+  type        = number
+  default     = 0
+}
+
+variable "cloud_connection_speed" {
+  description = "Speed in megabits per sec. Supported values are 50, 100, 200, 500, 1000, 2000, 5000, 10000. Required when creating new connection"
+  type        = number
+  default     = 5000
+}
+
+variable "configure_proxy" {
+  description = "Proxy is required to establish connectivity from PowerVS VSIs to the public internet. Do not configure proxy in this example by default."
+  type        = bool
+  default     = false
+}
+
+variable "configure_ntp_forwarder" {
+  description = "NTP is required to sync time over time server not reachable directly from PowerVS VSIs. Do not configure NTP forwarder in this example by default."
+  type        = bool
+  default     = false
+}
+
+variable "configure_dns_forwarder" {
+  description = "DNS is required to configure DNS resolution over server that is not reachable directly from PowerVS VSIs. Do not configure DNS forwarder in this example by default."
+  type        = bool
+  default     = false
+}
+
+variable "configure_nfs_server" {
+  description = "NFS server may be used to provide shared FS for PowerVS VSIs. Do not configure NFS server in this example by default."
+  type        = bool
+  default     = false
+}
+
+variable "resource_tags" {
+  type        = list(string)
+  description = "Optional list of tags to be added to created resources"
+  default     = []
+}
+
+variable "cloud_connection_gr" {
+  description = "Enable global routing for this cloud connection. Can be specified when creating new connection"
+  type        = bool
+  default     = true
+}
+
+variable "cloud_connection_metered" {
+  description = "Enable metered for this cloud connection. Can be specified when creating new connection"
+  type        = bool
+  default     = false
+}
+
+variable "prefix" {
+  description = "Prefix for resources which will be created."
+  type        = string
+  default     = "pvs"
+}
+
+variable "squid_proxy_config" {
+  description = "Configure SQUID proxy to use with IBM Cloud PowerVS instances."
+  type        = map(any)
+  default = {
+    squid_proxy_host_or_ip = null
+  }
+}
+
+variable "dns_forwarder_config" {
+  description = "Configure DNS forwarder to existing DNS service that is not reachable directly from PowerVS."
+  type        = map(any)
```

Comment:
my IDE gives out about this. Perhaps define the exact structure of the map instead of using `any`?

![Uploading image.png…]()